R/merge_sce_list.R

```diff
@@ -105,19 +131,59 @@ merge_sce_list <- function(sce_list = list(),
     warning("The provided `retain_coldata_cols` are not present in any SCEs.")
   }
 
+  # check that metadat columns are present
+  all_metadata_names <- purrr::map(sce_list, ~ names(metadata(.))) |>
+    unlist() |>
+    unname() |>
+    unique()
+
+  if (!(any(combine_metadata_cols %in% all_metadata_names))) {
+    warning("The provided `combine_metadata_cols` are not present in any SCEs.")
+  }
+
+
   # Prepare SCEs
   sce_list <- sce_list |>
     purrr::imap(prepare_sce_for_merge,
       batch_column = batch_column,
       cell_id_column = cell_id_column,
       shared_features = shared_features,
       retain_coldata_cols = retain_coldata_cols,
+      combine_metadata_cols = combine_metadata_cols,
       preserve_rowdata_cols = preserve_rowdata_cols
     )
 
   # Create the merged SCE from the processed list ------------------
   merged_sce <- do.call(cbind, sce_list)
 
+  # combine library metadata into a single data frame
+  metadata_list <- metadata(merged_sce)
+  metadata_to_keep <- metadata_list[!names(metadata_list) %in% c("library_metadata", "sample_metadata")]
+
+
+  all_library_metadata <- metadata_list[names(metadata_list) == "library_metadata"] |>
+    purrr::map(as.data.frame) |>
+    dplyr::bind_rows()
+
+  metadata_list <- c(
+    library_metadata = list(all_library_metadata),
+    metadata_to_keep
+  )
+
+  # if object has sample metadata then combine into a single data frame
+  if("sample_metadata" %in% names(metadata_list)){
+    all_sample_metadata <- metadata_list[names(metadata_list) == "sample_metadata"] |>
+      purrr::map(as.data.frame) |>
+      dplyr::bind_rows()
```

Comment:
Do we want to add `unique()` here in case we are combining multiple runs of the same sample?